### PR TITLE
Webpack 2

### PR DESCRIPTION
**Pull request by nickbalestra**
- [x] #346 Need to be merged first then we can rebase and add this in.

### CODE REVIEW

**Comment by matteofigus** [comment]
@nickbalestra I merged the other one. Can you rebase this? Cheers

File: package.json [hunk 1]
```diff
@@ -65,7 +65,6 @@
     "handlebars": "4.0.5",
     "infinite-loop-loader": "1.0.0",
     "jade": "1.11.0",
-    "json-loader": "0.5.4",
```
Comment (nickbalestra) [inline package.json]: https://webpack.js.org/guides/migrating/#json-loader-is-not-required-anymore

File: package.json [hunk 2]
```diff
@@ -84,6 +83,6 @@
     "uglify-js": "2.6.4",
     "underscore": "1.8.3",
     "watch": "0.19.1",
-    "webpack": "1.14.0"
+    "webpack": "2.2.0"
```
Comment (nickbalestra) [inline package.json]: If this version is not visible on npm, is cause it still behind the beta flag. Reason for this choice by the wp team is to speed up writing documentation. They will remove the beta flag once documentation is fully complete. But webpack itself is totally out of beta and stable

File: test/unit/cli-domain-package-server-script.js [hunk 3]
```diff
@@ -1,10 +1,6 @@
 'use strict';
 
 var expect = require('chai').expect;
-var injectr = require('injectr');
-var path = require('path');
-var sinon = require('sinon');
-var uglifyJs = require('uglify-js');
```
Comment (nickbalestra) [inline test/unit/cli-domain-package-server-script.js]: extra cleanup of unneded require for the new updated unit test.

File: npm-shrinkwrap.json [hunk 4]
```diff
@@ -3428,9 +3423,9 @@
       }
     },
     "webpack": {
-      "version": "1.14.0",
-      "from": "webpack@1.14.0",
-      "resolved": "https://registry.npmjs.org/webpack/-/webpack-1.14.0.tgz"
+      "version": "2.2.0",
+      "from": "webpack@2.2.0",
+      "resolved": "https://registry.npmjs.org/webpack/-/webpack-2.2.0.tgz"
```
Comment (matteofigus) [inline npm-shrinkwrap.json]: 🎉

File: src/cli/domain/package-server-script/bundle/config/index.js [hunk 5]
```diff
@@ -16,39 +16,39 @@ module.exports = function webpackConfigGenerator(params){
...
-              cacheDirectory: true,
-              'presets': [
-                [require.resolve('babel-preset-env'), {
-                  'targets': {
-                    'node': 4
-                  }
-                }]
-              ]
-            })
+          use: [
```
Comment (matteofigus) [inline src/cli/domain/package-server-script/bundle/config/index.js]: I really like the new syntax. Much cleaner

**Pull request merged**